src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -1313,24 +1336,16 @@ class UpdateNodes : public UpdateNodesBase {
 void Graph::InferDynamic(SyncInferRequest* request) {
     dnnl::stream stream(getEngine());
 
-    std::set<size_t> syncIndsWorkSet;
-    for (const auto& nodeIndx : syncNodesInds) {
-        syncIndsWorkSet.insert(nodeIndx.second);
-        //since sometimes we need to run the synchronization node  alone (for example in the case of internal dynamism)
-        //let's add another sync index after the sync point node
-        syncIndsWorkSet.insert(nodeIndx.second + 1);
-    }
-    syncIndsWorkSet.insert(executableGraphNodes.size());
+    auto getUpdateNodes = [](std::vector<NodePtr>& executableGraphNodes) -> std::unique_ptr<IUpdateNodes>{
+        if (parallel_get_max_threads() > 1)
+            return make_unique<UpdateNodes>(executableGraphNodes);
+        return make_unique<UpdateNodesSeq>(executableGraphNodes);
+    };
+    // @todo Can we reset UpdateNodes instead of creating for every inference?
+    auto updateNodes = getUpdateNodes(executableGraphNodes);
```

Comment:
```suggestion
    std::unique_ptr<IUpdateNodes> updateNodes = parallel_get_max_threads() < 2 ?make_unique<UpdateNodesSeq>(executableGraphNodes) : make_unique<UpdateNodes>(executableGraphNodes);
```
It is better to place more probable result on the else branch.
In the case of `unique_ptr` resetting doesn't make much sense as there is no control block.

Reply:
Tried this was well, ternary operator does not allow this.
As discussed, better to pass update strategy as a functor and make InferDynamic a template.
Will be more convenient to change in scope of a separate PR.
I will revert this change for now.